src/uct/tcp/sockcm/sockcm_ep.c

```diff
@@ -14,26 +14,242 @@
         } \
     } while (0)
 
+ucs_status_t uct_sockcm_ep_set_sock_id(uct_sockcm_iface_t *iface, uct_sockcm_ep_t *ep)
+{
+    ucs_status_t status;
+    struct sockaddr *dest_addr = NULL;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    ep->sock_id_ctx = ucs_malloc(sizeof(*ep->sock_id_ctx), "client sock_id_ctx");
+    if (ep->sock_id_ctx == NULL) {
+	status = UCS_ERR_NO_MEMORY;
+	goto out;
+    }
+
+    dest_addr = (struct sockaddr *) &(ep->remote_addr);
+
+    status = ucs_socket_create(dest_addr->sa_family, SOCK_STREAM,
+			       &ep->sock_id_ctx->sock_id);
+    if (status != UCS_OK) {
+	goto out_free;
+    }
+
+    ep->sock_id_ctx->ep = ep;
+    ucs_list_add_tail(&iface->used_sock_ids_list, &ep->sock_id_ctx->list);
+    ucs_debug("ep %p, new sock_id %d", ep, ep->sock_id_ctx->sock_id);
+    status = UCS_OK;
+    goto out;
+
+out_free:
+    ucs_free(ep->sock_id_ctx);
+out:
+    UCS_ASYNC_UNBLOCK(iface->super.worker->async);
+    return status;
+}
+
+ucs_status_t uct_sockcm_send_client_info(uct_sockcm_iface_t *iface, uct_sockcm_ep_t *ep)
+{
+    uct_sockcm_conn_param_t conn_param;
+    uct_sockcm_priv_data_hdr_t *hdr;
+    ssize_t sent_len = 0;
+    ssize_t recv_len = 0;
+    ssize_t offset = 0;
+    int connect_confirm = -1;
+
+    memset(&conn_param.private_data, 0, UCT_SOCKCM_PRIV_DATA_LEN);
+    hdr = &conn_param.hdr;
+
+    /* pack worker address into private data */
+    hdr->length = ep->pack_cb(ep->pack_cb_arg, "eth0",
```

Comment:
Device name isn't relevant for sockcm when listening address is set to `INADDR_ANY`. This may not be valid when a specific address is set. This will need some verification. Is there a test in `gtest` to check if things work as expected?

I've replaced with DEV_NAME_NULL for now.

Reply:
In gtest, we go over the available interfaces and for the sockaddr resources (interfaces) we take the ones that are IPoIB (and active and have a valid sa_family) 
https://github.com/openucx/ucx/blob/master/test/gtest/ucp/test_ucp_sockaddr.cc#L103
https://github.com/openucx/ucx/blob/master/test/gtest/uct/uct_test.cc#L233
So now that you are adding a new sockaddr transport, non IPoIB interfaces should be tested as well.